server/src/main/java/com/exacaster/lighter/backend/yarn/YarnBackend.java

```diff
@@ -57,6 +63,10 @@ private ApplicationState getState(String id) {
                 case KILLED:
                     return ApplicationState.KILLED;
             }
+        } catch (ApplicationNotFoundException e) {
+            LOG.error("Application not found in YARN: {}", id, e);
+            applicationStorage.findApplication(id)
+                    .ifPresent(application -> applicationStatusHandler.processApplicationError(application, e));
```

Comment:
I do not like this dependency between `ApplicationStatusHandler`, `ApplicationStorage` and `Backend` implementation.

It would be nicer, if we could handle these kind of errors outside of `Backend` scope. It would also be beneficial for possible future cases, when other errors will have to be handled in same manner.

We should wrap this specific YARN exception with some kind of Lighter exception, and catch it where this method is used. Is it possible?

Reply:
Discussed internally and addressed.